src/Components/Shared/SearchBar.tsx

```diff
@@ -0,0 +1,170 @@
+                            label="Album"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, album: !this.state.types.album } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.artist}
+                            checked={this.state.types.artist}
+                            label="Artist"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, artist: !this.state.types.artist } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.playlist}
+                            checked={this.state.types.playlist}
+                            label="Playlist"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, playlist: !this.state.types.playlist } })
+                            }}
+                        />
+                        <Form.Check
+                            inline
+                            disabled={!this.props.track}
+                            checked={this.state.types.track}
+                            label="Track"
+                            onChange={() => {
+                                this.setState({ types: { ...this.state.types, track: !this.state.types.track } })
+                            }}
+                        />
+                    </DropdownButton>
+                    <FormControl
+                        as="input"
+                        placeholder="Search"
+                        value={this.state.searchVal}
+                        onKeyDown={this.search}
+                        onChange={(e) => this.setState({ ...this.state, searchVal: e.target.value })}
+                    />
+                </InputGroup>
+                <Accordion>
+                    <Card>
+                        <Accordion.Toggle as={Card.Header} className="search-results-toggle__button" eventKey="0" onSelect={() => {}}>
+                            &#10225;
+                        </Accordion.Toggle>
+                        <Accordion.Collapse eventKey="0">
+                            <ul>{searchItems}</ul>
+                        </Accordion.Collapse>
+                    </Card>
+                </Accordion>
+            </div>
+        )
+    }
+
+    createQuery(): string {
+        let query = `q=${this.state.searchVal}`
+
+        if (
+            !Object.entries(this.state.types).every((k) => {
+                !k[1]
+            })
+        ) {
+            query += "&type="
+        }
+
+        for (const type in this.state.types) {
+            if (this.state.types[type as searchTypes]) {
+                query += type
```

Comment:
These params need a comma between them. I might modify this to have an array that you add the types to and then generate the string by doing `myTypeArr.join(",")`